parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh

```diff
@@ -103,18 +103,19 @@ EOF
 
 # CSE+VHD can dictate the containerd version, users don't care as long as it works
 installStandaloneContainerd() {
-    CONTAINERD_VERSION=$1    
+    local desiredVersion="${1:-}"
+    #e.g., desiredVersion will look like this 1.6.26-5.cm2
     # azure-built runtimes have a "+azure" suffix in their version strings (i.e 1.4.1+azure). remove that here.
     CURRENT_VERSION=$(containerd -version | cut -d " " -f 3 | sed 's|v||' | cut -d "+" -f 1)
     # v1.4.1 is our lowest supported version of containerd
     
-    if semverCompare ${CURRENT_VERSION:-"0.0.0"} ${CONTAINERD_VERSION}; then
-        echo "currently installed containerd version ${CURRENT_VERSION} is greater than (or equal to) target base version ${CONTAINERD_VERSION}. skipping installStandaloneContainerd."
+    if semverCompare ${CURRENT_VERSION:-"0.0.0"} ${desiredVersion}; then
+        echo "currently installed containerd version ${CURRENT_VERSION} is greater than (or equal to) target base version ${desiredVersion}. skipping installStandaloneContainerd."
     else
-        echo "installing containerd version ${CONTAINERD_VERSION}"
+        echo "installing containerd version ${desiredVersion}"
         removeContainerd
         # TODO: tie runc to r92 once that's possible on Mariner's pkg repo and if we're still using v1.linux shim
-        if ! dnf_install 30 1 600 moby-containerd; then
+        if ! dnf_install 30 1 600 "moby-containerd-${desiredVersion}"; then
```

Comment:
did you test to make sure specifying an actual version here works via build logs?

Reply:
yes I tested it manually. Let me paste the link.

Reply:
https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=96177599&view=logs&j=da6da462-43f6-5106-fd69-5730003ba730&t=274a1bb5-2aca-56be-d627-49f35f75c502&l=4725
```
==> azure-arm: + dnf_install 30 1 600 moby-containerd-1.6.26-5.cm2
==> azure-arm: + retries=30
==> azure-arm: + wait_sleep=1
==> azure-arm: + timeout=600
==> azure-arm: + shift
==> azure-arm: + shift
==> azure-arm: + shift
==> azure-arm: ++ seq 1 30
==> azure-arm: + for i in $(seq 1 $retries)
==> azure-arm: + dnf install -y moby-containerd-1.6.26-5.cm2
```